crates/wasi/Cargo.toml

```diff
@@ -12,9 +12,9 @@ keywords.workspace = true
 categories.workspace = true
 
 [dependencies]
-wasi-common = "2.0"
-wasi-cap-std-sync = "2.0"
-wiggle = { version = "2.0", default-features = false, features = ["wiggle_metadata"] }
+wasi-common = "17.0"
+wasi-cap-std-sync = "17.0"
+wiggle = { version = "17.0", default-features = false, features = ["wiggle_metadata"] }
```

Comment:
Same applies as in https://github.com/wasmi-labs/wasmi/pull/941/files#r1503430182.

We need to use `default-features = false` and `features = ["wiggle_metadata"]`